src/abstractdataframe/reshape.jl

```diff
@@ -185,50 +185,32 @@ long = stack(wide)
 wide0 = unstack(long)
 wide1 = unstack(long, :variable, :value)
 wide2 = unstack(long, :id, :variable, :value)
+wide3 = unstack(long, [:id, :a], :variable, :value)
 ```
 Note that there are some differences between the widened results above.
 
 """
-function unstack(df::AbstractDataFrame, rowkey::Int, colkey::Int, value::Int)
-    # `rowkey` integer indicating which column to place along rows
-    # `colkey` integer indicating which column to place along column headers
-    # `value` integer indicating which column has values
-    refkeycol = CategoricalArray{Union{eltype(df[rowkey]), Missing}}(df[rowkey])
-    # make sure we report only levels actually present in rowkey column
-    # this is consistent with how the other unstack method based on groupby works
-    droplevels!(refkeycol)
-    valuecol = df[value]
-    keycol = CategoricalArray{Union{eltype(df[colkey]), Missing}}(df[colkey])
-    Nrow = length(refkeycol.pool)
-    Ncol = length(keycol.pool)
-    payload = DataFrame(Any[similar_missing(valuecol, Nrow) for i in 1:Ncol], map(Symbol, levels(keycol)))
-    nowarning = true
-    for k in 1:nrow(df)
-        j = Int(CategoricalArrays.order(keycol.pool)[keycol.refs[k]])
-        i = Int(CategoricalArrays.order(refkeycol.pool)[refkeycol.refs[k]])
-        if i > 0 && j > 0
-            if nowarning && !ismissing(payload[j][i])
-                warn("Duplicate entries in unstack.")
-                nowarning = false
-            end
-            payload[j][i]  = valuecol[k]
-        end
-    end
-    levs = levels(refkeycol)
-    col = similar_missing(df[rowkey], length(levs))
-    insert!(payload, 1, copy!(col, levs), _names(df)[rowkey])
-end
+unstack(df::AbstractDataFrame, rowkey::Int, colkey::Int, value::Int) =
+    unstack(df, [rowkey], colkey, value)
 unstack(df::AbstractDataFrame, rowkey::ColumnIndex,
```

Comment:
Add a line break since you do the same below.

Reply:
fixed